### DISCUSSION TITLE
Label value  Coverage issues

### DISCUSSION THREAD

**Discussion by bobosmile1170556238** [prometheus/pushgateway#588] [Q&A] [answered]
# First
cat <<EOF | curl -X POST --data-binary @- http://127.0.0.1:9091/metrics/job/memory
memory{code="24", instance="A"} 1
EOF

# Second
cat <<EOF | curl -X PUT --data-binary @- http://127.0.0.1:9091/metrics/job/memory
memory{code="25", instance="B"} 5
EOF

Why does the second of these two examples overwrite the first? How can they coexist?

**Comment by beorn7** [comment, marked as answer]
As per the [documentation](https://github.com/prometheus/pushgateway/blob/master/README.md), the [PUT method](https://github.com/prometheus/pushgateway/blob/master/README.md#put-method) always replaces all metrics in the same group, while the [POST method](https://github.com/prometheus/pushgateway/blob/master/README.md#post-method) only replaces metrics with the same name (AKA the metric family). 

In your example, you are always pushing to the same group (`job/memory`), or in PromQL syntax `{job="memory"}`. Also, you only push a metric named `memory` (note that this is the metric name and has nothing to do with the `job` label value `memory` given in the grouping key). So in either case, you will _replace_ the previously pushed metrics.

If you want to push both of them in different pushes, and you want both of them to co-exist on the PGW, you have to push them into different groups. In your case, you could make the `code` label or the `instance` label (or both) part of the grouping key.

**Reply by bobosmile1170556238** [reply]
ok,thank you!
